arangod/RestServer/DatabaseFeature.h

```diff
@@ -27,17 +27,13 @@
 #include "Basics/Thread.h"
 #include "Containers/FlatHashMap.h"
 #include "Containers/FlatHashSet.h"
-#include "Metrics/Counter.h"
-#include "Metrics/Histogram.h"
-#include "Metrics/LogScale.h"
 #include "Replication2/Version.h"
 #include "RestServer/arangod.h"
 #include "Utils/DatabaseGuard.h"
 #include "Utils/VersionTracker.h"
 #include "VocBase/voc-types.h"
 #include "VocBase/Methods/Databases.h"
 
-#include <condition_variable>
```

Comment:
All this to just avoid condition_variable and metrics (for metrics it's possible just use Fwd.h)?

I'm not strong against such changes, but is it really worth?
git history will be broken for IOHeartbeatThread, same about backports

Reply:
Yes, but is it relevant that classes move between files? Especially an easy to locate class such as IOHeartbeatThread?
We have other PRs that change 200+ files and we don't complain.
The benefit of this PR IMO is not the we avoid including a few files in a central header file. That benefit will even be hard to measure. The main benefit IMO is that it improves modularization. If you need an IOHeartbeatThread, you now can have one (and only this) by including the appropriate header file. You don't get an IOHeartbeatThread as a (wanted or unwanted) side effect by including DatabaseFeature.h anymore. IMHO it should have been like this from the get-go, and this PR just rectifies it. (same for DatabaseManagerThread by the way, but I like to do changes in small iterations).